lib/infinispan.js

```diff
@@ -228,6 +228,14 @@
             .then(function() {
               logger.debugf('Started Infinispan %s', client);
               return client; // return client
+            }, function(reason){
+              transport.disconnect();
+
+              var error = reason;
+              if (typeof reason == "string") {
```

Comment:
Why is this needed exactly? Can't you just throw the error you got?

Reply:
Actually when I was rethrowing the error, I had some issues with ssl tests, but now it is fixed. Anyway, removed this part and just rethrowing the error.